src/win.rs

```diff
@@ -13,42 +13,107 @@ use std::{
 use windows_sys::Win32::{
-    Foundation::{ERROR_CANT_RESOLVE_FILENAME, HANDLE, TRUE, ERROR_DIRECTORY, ERROR_NOT_A_REPARSE_POINT, ERROR_INVALID_PARAMETER, ERROR_NO_MORE_FILES},
+    Foundation::{
+        ERROR_CANT_RESOLVE_FILENAME, ERROR_DIRECTORY, ERROR_INVALID_PARAMETER,
+        ERROR_NOT_A_REPARSE_POINT, ERROR_NO_MORE_FILES, HANDLE, TRUE,
+    },
+    Security::{SECURITY_DESCRIPTOR, SECURITY_QUALITY_OF_SERVICE},
     Storage::FileSystem::{
         FileDispositionInfo, FileIdBothDirectoryInfo, FileIdBothDirectoryRestartInfo,
         GetFileInformationByHandleEx, NtCreateFile, SetFileInformationByHandle, DELETE,
-        FILE_ATTRIBUTE_NORMAL, FILE_GENERIC_WRITE, FILE_INFO_BY_HANDLE_CLASS, FILE_LIST_DIRECTORY,
-        FILE_READ_ATTRIBUTES, FILE_SHARE_DELETE, FILE_SHARE_READ, FILE_SHARE_WRITE, FILE_TRAVERSE,
-        FILE_WRITE_ATTRIBUTES, FILE_WRITE_DATA, FILE_DISPOSITION_INFO, FILE_ID_BOTH_DIR_INFO, SYNCHRONIZE, MAXIMUM_REPARSE_DATA_BUFFER_SIZE,
+        FILE_ATTRIBUTE_NORMAL, FILE_CREATE, FILE_DISPOSITION_INFO, FILE_GENERIC_WRITE,
+        FILE_ID_BOTH_DIR_INFO, FILE_INFO_BY_HANDLE_CLASS, FILE_LIST_DIRECTORY, FILE_OPEN,
+        FILE_OPEN_IF, FILE_OVERWRITE_IF, FILE_READ_ATTRIBUTES, FILE_SHARE_DELETE, FILE_SHARE_READ,
+        FILE_SHARE_WRITE, FILE_TRAVERSE, FILE_WRITE_ATTRIBUTES, FILE_WRITE_DATA,
+        MAXIMUM_REPARSE_DATA_BUFFER_SIZE, SYNCHRONIZE,
     },
     System::{
-        SystemServices::{GENERIC_READ, GENERIC_WRITE, IO_REPARSE_TAG_SYMLINK, IO_REPARSE_TAG_MOUNT_POINT},
-        WindowsProgramming::OBJECT_ATTRIBUTES,
-        IO::DeviceIoControl, Kernel::OBJ_CASE_INSENSITIVE, Ioctl::{FSCTL_SET_REPARSE_POINT, FSCTL_GET_REPARSE_POINT},
-    }, Security::{SECURITY_DESCRIPTOR, SECURITY_QUALITY_OF_SERVICE},
+        Ioctl::{FSCTL_GET_REPARSE_POINT, FSCTL_SET_REPARSE_POINT},
+        Kernel::OBJ_CASE_INSENSITIVE,
+        SystemServices::{
+            GENERIC_READ, GENERIC_WRITE, IO_REPARSE_TAG_MOUNT_POINT, IO_REPARSE_TAG_SYMLINK,
+        },
+        WindowsProgramming::{
+            FILE_CREATED, FILE_DIRECTORY_FILE, FILE_DOES_NOT_EXIST, FILE_EXISTS, FILE_OPENED,
+            FILE_OPEN_REPARSE_POINT, FILE_OVERWRITTEN, FILE_SUPERSEDED,
+            FILE_SYNCHRONOUS_IO_NONALERT, OBJECT_ATTRIBUTES,
+        },
+        IO::DeviceIoControl,
+    },
 };
 
 use crate::{LinkEntryType, OpenOptions, OpenOptionsWriteMode};
 
+use exports::SECURITY_CONTEXT_TRACKING_MODE;
+
+use self::reparse_definitions::{
+    REPARSE_DATA_BUFFER_u_SymbolicLinkReparseBuffer, REPARSE_DATA_BUFFER,
+};
+
 pub mod exports {
     pub use super::OpenOptionsExt;
-    #[doc(no_inline)]
-    pub use winapi::um::winnt::SECURITY_CONTEXT_TRACKING_MODE;
+    // SECURITY_CONTEXT_TRACKING_MODE is not available in windows_sys yet, but it's a pretty simple definition
+    // so in order to maintain API compatibility we'll replicate it here.
+    #[allow(non_camel_case_types)]
+    pub type SECURITY_CONTEXT_TRACKING_MODE = u8;
```

Comment:
is there a tracking bug for it?

Reply:
Filing one now

Reply:
Filed and linked https://github.com/microsoft/win32metadata/issues/1464